redis_enterprise/datadog_checks/redis_enterprise/redis_enterprise.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any  # noqa: F401
+
+from datadog_checks.base import AgentCheck, ConfigurationError, OpenMetricsBaseCheckV2  # noqa: F401
+
+
+class RedisEnterpriseCheck(OpenMetricsBaseCheckV2):
+
+    # This will be the prefix of every metric and service check the integration sends
+    # __NAMESPACE__ = 'redis_enterprise'
+
+    def __init__(self, name, init_config, instances):
+
+        super(RedisEnterpriseCheck, self).__init__(name, init_config, instances)
+
+        print(f'config: {init_config}')
+        print(f'instances: {instances}')
+
+        # Use self.instance to read the check configuration
+        self.endpoint = self.instance.get('openmetrics_endpoint')
+        if self.endpoint is None:
+            raise ConfigurationError("Unable to find openmetrics_endpoint in config file.")
+
+    def can_connect(self, hostname=None, message=None, tags=None):
+        print(f'hostname: {hostname}, message: {message}, tags: {tags}')
+        return False
+
+    def check(self, instance):
+        # type: (Any) -> None
+
+        # If your check ran successfully, you can send the status.
+        # More info at
+        # https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck.service_check
+        # self.service_check("can_connect", AgentCheck.OK)
+
+        # If it didn't then it should send a critical service check
+        # self.service_check("can_connect", AgentCheck.CRITICAL)
+
+        try:
+            super().check(instance)
```

Comment:
You have a `metadata.csv` defined with a limited set of metrics. This means those metrics should be collected with minimal configuration necessary from your users.

However the way your implementation works right now, your users **have to** configure a set of metrics to collect.

1. More manual configuration makes user experience worse.
2. We open the door for users to configure metrics that are not declared in the `metadata.csv`.

I would recommend you check out [this implementation for example](https://github.com/DataDog/integrations-core/blob/79fbbbc3958be841c2ebb0c2815f51403970b446/dcgm/datadog_checks/dcgm/check.py#L9) to see how to collect a specific consistent set of metrics.

Reply:
Thanks, this was an open question I had. I will switch to this style.

Reply:
There is the question of typing, however. Previously we had noted that certain metrics were not available unless they had a type/type_overrides declaration. How do we accomplish this with this metrics style?

Reply:
eg.
      - kong_latency:
          name: latency
          type: counter

Reply:
nevermind

'kong_latency': {
                'name': 'latency',
                'type': 'counter',
            },

Reply:
Right, you can provide the type overrides in the metric mapping.

Think of it as hard-coding a config into the integration's source code, the datastructure is the same.